### ISSUE TITLE
Delete single feeds

### ISSUE THREAD

**Issue by balcy** [florisluiten/rssreader#4]
It would be good, if single feeds could be removed from the list

**Comment by florisluiten** [comment]
Thank you for taking time to give me feedback.

I've never had to delete a feed myself, but I agree that such an option should exists.

I'm not sure how the deleting should work though. In the normal Ubuntu Touch UI one would swipe an item to the right and a "BIN" icon would appear.
I'm not sure if I'm able to do this with the HTML Toolkit I'm using for the RSS Reader, but honestly haven't look a while at the documantation. I'll have to look into it.

I'll keep updates about this feature in this issue, so check back in a while.

**Comment by balcy** [comment]
Hi Floris,
ah OK I first thought the app was written in QML...
Maybe a long press on the feed title with a confirmation dialog / context menu could be used instead.
The app is still great and I enjoy using it ! The deleting feature is rather something for the future / long term : )
thank you for creating the app and kind regards,
Chris

**Comment by florisluiten** [comment]
Thank you for your kind words Chris, that makes my day!
Features for the app are always welcome. I've created the reader for myself, and have no idea how other users use it. It helps me greatly by feedback of users (you) to find ways to improve the app for everyone. 

I agree that a long press should be a good alternative. Although I prefer to stay as close to the Ubuntu Style guides as possible. Also, I think for the user it should not matter if the application is written in QML or HTML. The user experience should always be consistent.

I've implemented a third-party library for the dragging and altered the reader in order to allow for the "swipe to delete" action. It works well, if i may say so myself. Please update to version 0.1.7 (which I hope to release within the next 24 hours) to see the changes. If you have any more feedback about the deleting of the feed, feel free to add your feedback to this issue.

**Comment by balcy** [comment]
Hi Floris ! 
Thank you for implementing the feature !

Just one UI question and one issue I have found: To be consistent with other apps, should it just show the trash icon after doing the swipe, and then a click on the trash icon trigger the delete (instead of the swipe itself) ?

The other issue was, that I had a situation, that deleting one feed actually deletes 2, I try describing it:
- add one feed (1)
- add another feed (2)
- swipe (1) to trigger the delete action
- select cancel
- swipe (1) to trigger the delete action
- this time confirm the delete
  --> feeds (1) and (2) are deleted

The feature itself is very useful : )
kind regards,
Chris

**Comment by florisluiten** [comment]
Hi Chris,

Thanks again for your feedback. You are absolutely right about the swipe action. I've changed this, to be consistent with the Ubuntu UI. 

The double deleting issue was somewhat more difficult to find, the debugging didn't really help me. Eventually I found that the behaviour was due to the "cancel" button not really working. The dialog disappeared, but the action was never cancelled. As similar issue happened in the "Add new feed" dialog.

I fixed both issues, so look for update 0.1.8 coming soon (again, hoping to release within the next 24 hours).
